torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1433,6 +1433,37 @@ class ShapePropagator {
           dtype, device, dim, /*requires_grad=*/c10::nullopt)};
     };
 
+    static const auto factory_like_with_ndim = [](Node* node,
+                                                  int dim) -> type_vec_t {
+      at::optional<IValue> maybe_layout_option = node->get(attr::layout);
+      if (!maybe_layout_option)
+        return {};
+
+      at::optional<IValue> maybe_device_option = node->get(attr::device);
+      if (!maybe_device_option)
+        return {};
+
+      auto tt = node->input(0)->type()->expect<TensorType>();
+
+      auto in_type = tt->scalarType();
+      auto in_dev = tt->device();
+
+      if (!maybe_device_option->isNone()) {
+        in_dev = maybe_device_option->toDevice();
+      }
+
+      at::optional<IValue> maybe_dtype_option = node->get(attr::dtype);
+      if (!maybe_dtype_option)
```

Comment:
Can you move this up to the other checking of `at::optional<IValue> maybe_device_option = node->get(attr::device);` (instead of after `in_dev = maybe_device_option->toDevice`)? this would be easier to follow if all of the checking followed the same pattern.

Reply:
will do.